tests/template_tests/syntax_tests/test_i18n.py

```diff
@@ -435,6 +435,24 @@ def test_i18n38_2(self):
             'fr: French/français/francouzsky bidi=False; '
         )
 
+    # blocktrans tag with asvar
+    @setup({'i18n39': '{% load i18n %}{% blocktrans asvar page_not_found %}Page not found{% endblocktrans %}{{ page_not_found }}'})
+    def test_i18n39(self):
+        with translation.override('de'):
+            output = self.engine.render_to_string('i18n39')
+        self.assertEqual(output, 'Seite nicht gefunden')
+
+    @setup({'i18n40': '{% load i18n %}{% trans "Page not found" as pg_404 %}{% blocktrans with page_not_found=pg_404 asvar output %}Error: {{ page_not_found }}{% endblocktrans %}'})
+    def test_i18n40(self):
+        output = self.engine.render_to_string('i18n40')
+        self.assertEqual(output, '')
+
+    @setup({'i18n41': '{% load i18n %}{% trans "Page not found" as pg_404 %}{% blocktrans with page_not_found=pg_404 asvar output %}Error: {{ page_not_found }}{% endblocktrans %}{{ output }}'})
```

Comment:
Can you add an indication character right before and after `{{ output }}`, just to make sure that the output really comes at the right place. I.e.: `'{% endblocktrans %}>{{ output }}<'`